### ISSUE TITLE
Can't display related field

### ISSUE THREAD

**Issue by misterx** [directus/directus#15298]
### Describe the Bug

Related table displays incorrect data when data structure is `record->o2m->m2o->field`.
Looks as UI bug, because requests and responses are correct.

### To Reproduce

I'm using the following structure:

Invoices
----------
items -> o2m -> InvoiceItem

Invoice Items
----------
invoice_id -> m2o -> Invoice
developer-> m2o -> Developer

Developer
----------
name

UI configuration
-----------------
I've configured `Invoices.items` interface for displaying `Developer.Name` and `Developer.ID` but  `InvoiceItem.ID` is displayed instead of  `Developer.ID` also `Developer.Name` is null

Screenshots
--------------
`Invoice.items` field configuration
![image](https://user-images.githubusercontent.com/216417/187070240-d3ffabb6-d794-458a-9525-a765d78a7122.png)
`Invoice.items` interface configuration
![image](https://user-images.githubusercontent.com/216417/187070269-4002fc92-fa7b-4388-9470-a289802e0298.png)
`InvoiceItem.developer` field configuration
![image](https://user-images.githubusercontent.com/216417/187070375-0c8f4e5e-16a1-4b11-9193-b19305552274.png)
Invoice example
![image](https://user-images.githubusercontent.com/216417/187070428-c951ecc2-f938-4a33-ad26-f0e1b323e122.png)

Requests and response
------------------
When invoice opened, UI fetch related data from `/items/InvoiceItem?fields[]=developer.id&fields[]=developer.name&fields[]=minutes&fields[]=id&filter=%7B%22_and%22:[%7B%22invoice_id%22:%2233b780e1-340d-457c-9e8c-18a712091fdf%22%7D]%7D&page=1&limit=15`

Response:

```json
{
    "data": [
        {
            "minutes": 5,
            "id": "6e9c909b-0c78-4a1b-af55-46fb7114fe9e",
            "developer": {
                "id": "2171251b-542c-47b8-aa99-8161cf56db38",
                "name": "Developer 1"
            }
        }
    ]
}
```

### What version of Directus are you using?

9.16.1

### What version of Node.js are you using?

16.17.0

### What database are you using?

Postgres 14

### What browser are you using?

Chrome

### How are you deploying Directus?

Docker

**Comment by azrikahar** [comment]
This should be the same issue as #15154. Tested the fix #15155 and can confirm it fixes this.

### Tested interface configuration

![chrome_az7OyqIy7h](https://user-images.githubusercontent.com/42867097/187129175-1c1bf58d-d13e-4d9d-a48f-b98f8aa750df.png)

### Before the fix

![chrome_cZn7DtSEvP](https://user-images.githubusercontent.com/42867097/187129165-cb5c762a-eb86-4adb-b611-6033afd2f00b.png)

### After the fix

![chrome_S2fhJ4yUpV](https://user-images.githubusercontent.com/42867097/187129161-03fe9aad-cc74-4b34-ae45-f2ae6249eaaf.png)

**Comment by enessefak** [comment]
Hi Directus 9.23.1 version visible problem

<img width="632" alt="image" src="https://user-images.githubusercontent.com/10434186/224703247-308c4aeb-ed4b-47a1-9769-50582975568e.png">

<img width="1201" alt="image" src="https://user-images.githubusercontent.com/10434186/224703391-66570bcd-81a7-4253-99a5-0abaa27342f5.png">
